test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -415,9 +419,11 @@ void testBCTest(json_spirit::mObject const& _o)
 {
 	string testName = TestOutputHelper::get().testName();
 	TestBlock genesisBlock(_o.at("genesisBlockHeader").get_obj(), _o.at("pre").get_obj());
-	TestBlockChain blockchain(genesisBlock);
 
-	TestBlockChain testChain(genesisBlock);
+    TestBlockChain::MiningType miningType = getMiningType(_o);
+    eth::IncludeSeal includeSeal = (miningType == TestBlockChain::MiningType::Ethash || miningType == TestBlockChain::MiningType::Default) ? WithSeal : WithoutSeal;
```

Comment:
Define a variable closer to where you use it.

Reply:
the declaration must be before the line 
```
for (auto const& bl: _o.at("blocks").get_array())
```

Reply:
Right, but not at the top of the function

Reply:
do you want me to move it 4 lines below ? 
just what is the difference?

Reply:
Ideally of course I want you to never ever write loops 110 lines long and functions 150 lines long. Giant scope of a variable is only a consequence of this...